stageleft/src/type_name.rs

```diff
@@ -11,14 +11,106 @@ struct RewriteAlloc {
+                    .into_iter()
+                    .chain(i.segments.iter().skip(3).cloned()),
+                ),
+            };
+        } else if i.segments.iter().take(3).collect::<Vec<_>>()
+            == vec![
+                &syn::PathSegment::from(syn::Ident::new("core", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("iter", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("adapters", Span::call_site())),
+            ]
+        {
+            *i = syn::Path {
+                leading_colon: i.leading_colon,
+                segments: syn::punctuated::Punctuated::from_iter(
+                    vec![
+                        syn::PathSegment::from(syn::Ident::new("std", Span::call_site())),
+                        syn::PathSegment::from(syn::Ident::new("iter", Span::call_site())),
+                    ]
+                    .into_iter()
+                    .chain(i.segments.iter().skip(4).cloned()),
+                ),
+            };
+        } else if i.segments.iter().take(4).collect::<Vec<_>>()
+            == vec![
+                &syn::PathSegment::from(syn::Ident::new("std", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("collections", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("hash", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("map", Span::call_site())),
+            ]
+        {
+            *i = syn::Path {
+                leading_colon: i.leading_colon,
+                segments: syn::punctuated::Punctuated::from_iter(
+                    vec![
+                        syn::PathSegment::from(syn::Ident::new("std", Span::call_site())),
+                        syn::PathSegment::from(syn::Ident::new("collections", Span::call_site())),
+                        syn::PathSegment::from(syn::Ident::new("hash_map", Span::call_site())),
+                    ]
+                    .into_iter()
+                    .chain(i.segments.iter().skip(4).cloned()),
+                ),
+            };
+        } else if i.segments.iter().take(3).collect::<Vec<_>>()
+            == vec![
+                &syn::PathSegment::from(syn::Ident::new("tokio", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("time", Span::call_site())),
+                &syn::PathSegment::from(syn::Ident::new("instant", Span::call_site())),
+            ]
+        {
+            *i = syn::Path {
+                leading_colon: i.leading_colon,
+                segments: syn::punctuated::Punctuated::from_iter(
+                    vec![
+                        syn::PathSegment::from(syn::Ident::new("tokio", Span::call_site())),
+                        syn::PathSegment::from(syn::Ident::new("time", Span::call_site())),
+                    ]
+                    .into_iter()
+                    .chain(i.segments.iter().skip(3).cloned()),
+                ),
+            };
```

Comment:
whats going on here?

Reply:
So we use `std::any::type_name` to get the concrete type of function inputs/outputs so that in the generated code we do not have to rely on type inference behavior. But `type_name` will return the canonicalized path to the original definition, which could be in a private module. So as a (temporary) hack around this, we have some rewrite rules for common types that have this issue.